pkg/api/v1/defaults.go

```diff
@@ -151,6 +152,34 @@ func setDefaults_DiskFromMachineType(obj *VirtualMachine) {
 	}
 }
 
+func setDefaults_NetworkInterface(obj *VirtualMachine) {
+	interfaces := obj.Spec.Domain.Devices.Interfaces
+	if len(interfaces) == 0 {
```

Comment:
I would do that differently:

 * First check if `networks` is empty or does not contain a pod network.
 * If one of the above is true:
      -  Add the network with a specific name.
      - Add an interface with the same name.
 * Else, do nothing.

That seems to be the more correct approach.